akka-remote/src/main/scala/akka/remote/transport/AkkaProtocolTransport.scala

```diff
@@ -394,8 +393,10 @@ private[transport] class ProtocolStateActor(
           // After receiving Disassociate we MUST NOT send back a Disassociate (loop)
           stop(FSM.Failure(info))
 
-        case _ ⇒
+        case msg ⇒
           // Expected handshake to be finished, dropping connection
+          if (log.isDebugEnabled)
+            log.debug("Sending disassociate to {} because unexpected message was received during handshake {}", wrappedHandle, msg)
```

Comment:
I think we should avoid full `toString` of messages, also on debug level, to not risk bad performance (or even OOME). Use the class name.